### PR TITLE
Add pallet-maintenance-mode and pallet-migrations to tanssi and both templates

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Added tests to the common folder so that they run in both tanssi and frontier template.

### CODE REVIEW

**Comment by girazoki** [comment]
I think we also want to add it to the templates. Templates will also likely need migrations.

**Comment by girazoki** [comment]
We might also want to incorporate maintenance-mode, now that we are already bringing moonkit stuff. I know we had this in the list but and the sooner we incorporate it, the better I think

File: container-chains/templates/frontier/runtime/src/lib.rs [hunk 1]
```diff
@@ -516,6 +520,101 @@ impl pallet_utility::Config for Runtime {
...
+    type MigrationsList = ();
+    type XcmExecutionManager = ();
+}
+
+/// Maintenance mode Call filter
+pub struct MaintenanceFilter;
+impl Contains<RuntimeCall> for MaintenanceFilter {
+    fn contains(c: &RuntimeCall) -> bool {
+        match c {
+            RuntimeCall::Balances(_) => false,
```
Comment (girazoki) [inline container-chains/templates/frontier/runtime/src/lib.rs]: We probably want to prevent the evm too here

File: runtime/dancebox/src/lib.rs [hunk 2]
```diff
@@ -650,6 +653,108 @@ impl pallet_proxy::Config for Runtime {
...
+    type MigrationsList = ();
+    type XcmExecutionManager = ();
+}
+
+/// Maintenance mode Call filter
+pub struct MaintenanceFilter;
+impl Contains<RuntimeCall> for MaintenanceFilter {
+    fn contains(c: &RuntimeCall) -> bool {
+        match c {
+            RuntimeCall::Balances(_) => false,
```
Comment (girazoki) [inline runtime/dancebox/src/lib.rs:667]: is proxy not in the list of moonbeam?

Comment (tmpolaczyk) [reply runtime/dancebox/src/lib.rs:667]: Yes, I may have accidentally removed it

File: container-chains/templates/frontier/runtime/src/lib.rs (hunk 1 above)
Comment (tmpolaczyk) [reply container-chains/templates/frontier/runtime/src/lib.rs]: Added evm but accidentally removed balances, will add it back

File: runtime/dancebox/src/lib.rs (hunk 2 above)
Comment (tmpolaczyk) [reply runtime/dancebox/src/lib.rs:667]: Actually no, proxy is not in the MaintenanceFilter list.

File: runtime/dancebox/src/lib.rs [hunk 3]
```diff
@@ -650,6 +653,108 @@ impl pallet_proxy::Config for Runtime {
...
+                pallet_proxy::Call::kill_pure { .. } => false,
+                _ => true,
+            },
+            _ => true,
+        }
+    }
+}
+
+/// The hooks we want to run in Maintenance Mode
+pub struct MaintenanceHooks;
```
Comment (girazoki) [inline runtime/dancebox/src/lib.rs:695]: You can probably avoid this until we have XCM.

JUst set `MaintenanceExecutiveHooks=AllPalletsWithSystem`

Comment (girazoki) [reply runtime/dancebox/src/lib.rs:695]: same for template runtimes

**Review by girazoki — APPROVED** [review]

**Pull request merged**